### PR TITLE
feat: quick look support

### PR DESCRIPTION
**Pull request by sakul-budhathoki**
# Overview

Added support for a quick look for react-native-share. 

# Test Plan

Simulator Screenshot - iPhone 15 Pro - 2024-05-29 at 17 02 01

https://github.com/react-native-share/react-native-share/assets/13379067/667ebc6d-86b5-42de-869b-b0364686ff6b

Test code can be found at: https://github.com/sakul-budhathoki/react-native-share/tree/feat/quick-look-support-test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,7 +57,8 @@
     "react-native-windows": "^0.72.19",
     "react-test-renderer": "^18.2.0",
     "semantic-release": "^22.0.7",
-    "typescript": "^5.2.2"
+    "typescript": "^5.2.2",
+    "react-native-fs": "^2.20.0"
```
Comment (EstebanFuentealba) [inline package.json:61]: **react-native-fs** I believe it's best to avoid dependencies on other projects and let the end user decide which package to use.

Comment (sakul-budhathoki) [reply package.json:61]: Ok. Thanks. It's used only to demo the implementation.

File: example/App.js [hunk 2]
```diff
@@ -20,16 +20,37 @@ import {
...
 import pdfBase64 from './images/pdfBase64';
 import {video} from './videos/videoBase64';
 
+const LOCAL_PDF_FILE_NAME = 'pdf-test.pdf';
+const LOCAL_PDF_TEST_FILE_PATH = `${RNFS.DocumentDirectoryPath}/${LOCAL_PDF_FILE_NAME}`;
+
+const downloadPDF = async () => {
+  try {
+    const options = {
+      fromUrl: 'https://www.orimi.com/pdf-test.pdf',
```
[the remark this answers is not in the document]
Comment (sakul-budhathoki) [reply example/App.js:35]: Ok. Thanks. This code will be removed before merge; as I will be removing `react-native-fs`.

**Comment by robertscherer** [comment]
Hey @MateusAndrade , is there something we can help with to get this one merged?

**Comment by MateusAndrade** [comment]
> Hey @MateusAndrade , is there something we can help with to get this one merged?

Looking at it, we are missing some changes with the javascript/typescript part as well as for docs.

**Pull request closed without merging**